lib/manager/terraform/lockfile/hash.ts

```diff
@@ -0,0 +1,170 @@
+import crypto from 'crypto';
+import extract from 'extract-zip';
+import pMap from 'p-map';
+import { join } from 'upath';
+import { TerraformProviderDatasource } from '../../../datasource/terraform-provider';
+import type {
+  TerraformBuild,
+  VersionDetailResponse,
+} from '../../../datasource/terraform-provider/types';
+import { logger } from '../../../logger';
+import * as packageCache from '../../../util/cache/package';
+import * as fs from '../../../util/fs';
+import { Http } from '../../../util/http';
+import { getCacheDir, repositoryRegex } from './util';
+
+const http = new Http(TerraformProviderDatasource.id);
+const hashCacheTTL = 10080; // in seconds == 1 week
+const terraformProviderDatasource = new TerraformProviderDatasource();
+
+export async function hashFiles(files: string[]): Promise<string> {
+  const rootHash = crypto.createHash('sha256');
+
+  for (const file of files) {
+    // build for every file a line looking like "aaaaaaaaaaaaaaa  file.txt\n"
+    const hash = crypto.createHash('sha256');
+
+    // a sha256sum displayed as lowercase hex string to root hash
+    const fileBuffer = await fs.readFile(file);
+    hash.update(fileBuffer);
+    hash.end();
+    const data = hash.read();
+    rootHash.update(data.toString('hex'));
+
+    // add double space, the filename and a new line char
+    rootHash.update('  ');
+    const fileName = file.replace(/^.*[\\/]/, '');
+    rootHash.update(fileName);
+    rootHash.update('\n');
+  }
+
+  rootHash.end();
+  const rootData = rootHash.read();
+  const result: string = rootData.toString('base64');
+  return result;
+}
+
+export async function hashOfZipContent(
+  zipFilePath: string,
+  extractPath: string
+): Promise<string> {
+  await extract(zipFilePath, { dir: extractPath });
+  const files = await fs.readdir(extractPath);
+  // the h1 hashing algorithms requires that the files are sorted by filename
+  const sortedFiles = files.sort((a, b) => a.localeCompare(b));
+  const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+
+  const result = await hashFiles(filesWithPath);
```

Comment:
@secustor Here the await was missing, so the below delete occured while hashing was running. Thankfully it failed in ci 😅

Reply:
Strange that this only happened on windows.

Reply:
![giphy-3](https://user-images.githubusercontent.com/6311784/122360948-19fc4680-cf57-11eb-8ed5-36d99ae5a1d7.gif)

Reply:
it's not happening on linux, because there the test hashes are computed too fast, i also couldn't reproduce on my local windows, it's to fast too 😅

Reply:
yeah totally my mistake **shame**